docs/DeveloperGuide.md

```diff
@@ -1,34 +1,544 @@
+even after the program is turned off. 
+
+### Architecture
+
+![Architecture](images/diagrams/Architecture.png)  <br>
+
+The *Architecture Diagram* shown above explains the high-level design of SmortHomeBot Application. Given below is a brief overview of each component.
+
+`Main` is responsible for initializing other components in correct sequence, and connects them up with each other.
+
+`Commons` represents a collection of classes used by multiple components. 
+
+The rest of the App consists of four components.
+* `Ui`: The user interface where user can enter instructions and view output.
+* `Logic` The command executor which consists of,
+   * `Paser`: Extract the keyword from user input
+   * `Commands`: Execute the specific command according to the keyword
+* `Data` Holds the data in-app-memory while the program is running. 
+* `Storage` Reads and writes data from and to a text file.
+
+How the architecture components interact with each other
+
+The *Sequence Diagram* below shows how the components interact with each other for the scenario where the user issues the command `create br1`.
+
+![UI Class](images/diagrams/Sequence_overview.png)  <br>
+
+The sections below give more details of each component.
+
+### Ui Component
+
+The class diagram of `TextUi` is shown below together with `Main`:
+
+![UI Class](images/diagrams/ClassDiagram_UI.png)  <br>
+
+The sequence diagram of `TextUi` is shown below:
+
+![UI Sequence](images/diagrams/Sequence_TextUi.png)  <br>
+
+**API** : [TextUI.java](https://github.com/AY2021S1-CS2113-T14-1/tp/blob/master/src/main/java/seedu/smarthomebot/ui/TextUi.java)
+
+The UI component which consists of TextUi,
+* Prompt commands from the user.
+* Execute user commands using the *Logic component*.
+* Displays information based on changes to *Data*.
+* Prints the SmartHomeBot appliance in a well-formatted form.
+
+### Logic Component
+
+The class diagram of `Logic Component` is shown below together with an example:
+
+![Logic Component](images/diagrams/Sequence_LogicComponent.png)  <br>
+
+**API** : [Logic](https://github.com/AY2021S1-CS2113-T14-1/tp/tree/master/src/main/java/seedu/smarthomebot/logic)
+
+1. Logic uses the **Parser class** to parse the user command.
+2. This results in a **Command object**.
+3. The command execution can affect the *Data* (e.g. adding a new location “br1” into its **LocationList**).
+4. The result of the command execution is encapsulated as a **CommandResult object** which is passed back to the Ui.
+5. In addition, the **CommandResult object** can also instruct the *Ui* to perform certain actions, such as displaying help to the user.
+6. Shown above is the Sequence Diagram for interactions within the *Logic component* for the when user enter "create br1" and how the API call.
+
+### Data Component
+
+The class diagram of `Data Component` is shown below:
+
+![Data Component](images/diagrams/ClassDiagram_DataOverview.png)  <br>
```

Comment:
In the composition relationships, is the multiplicity for the "whole" object necessary?